deployment/ccip/changeset/cs_chain_contracts.go

```diff
@@ -351,6 +353,283 @@ func UpdateOnRampsDestsChangeset(e deployment.Environment, cfg UpdateOnRampDests
+			ReentrancyGuardEntered: false,
+			MessageInterceptor:     update.MessageInterceptor,
+			FeeAggregator:          update.FeeAggregator,
+			AllowlistAdmin:         update.AllowlistAdmin,
+		})
+		if err != nil {
+			return deployment.ChangesetOutput{}, err
+		}
+		if cfg.MCMS == nil {
+			if _, err := deployment.ConfirmIfNoError(e.Chains[chainSel], tx, err); err != nil {
+				return deployment.ChangesetOutput{}, err
+			}
+		} else {
+			batches = append(batches, timelock.BatchChainOperation{
+				ChainIdentifier: mcms.ChainIdentifier(chainSel),
+				Batch: []mcms.Operation{
+					{
+						To:    onRamp.Address(),
+						Data:  tx.Data(),
+						Value: big.NewInt(0),
+					},
+				},
+			})
+			timelocks[chainSel] = onchain.Chains[chainSel].Timelock.Address()
+			proposers[chainSel] = onchain.Chains[chainSel].ProposerMcm
+		}
+	}
+	if cfg.MCMS == nil {
+		return deployment.ChangesetOutput{}, nil
+	}
+	proposal, err := proposalutils.BuildProposalFromBatches(
+		timelocks, proposers, batches,
+		"update onramp dynamic config",
+		cfg.MCMS.MinDelay)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	return deployment.ChangesetOutput{
+		Proposals: []timelock.MCMSWithTimelockProposal{*proposal},
+	}, nil
+}
+
+type UpdateOnRampAllowListConfig struct {
+	// UpdatesByChain is a mapping of source -> dest -> update.
+	UpdatesByChain map[uint64]map[uint64]OnRampAllowListUpdate
+	// Disallow mixing MCMS/non-MCMS per chain for simplicity.
+	// (can still be achieved by calling this function multiple times)
+	MCMS *MCMSConfig
+}
+
+type OnRampAllowListUpdate struct {
+	AllowListEnabled          bool
+	AddedAllowlistedSenders   []common.Address
+	RemovedAllowlistedSenders []common.Address
+}
+
+func (cfg UpdateOnRampAllowListConfig) Validate() error {
+	for srcSel, updates := range cfg.UpdatesByChain {
+		for destSel, update := range updates {
+			if len(update.AddedAllowlistedSenders) > 0 && !update.AllowListEnabled {
+				return fmt.Errorf("can't allowlist senders with disabled allowlist for src=%d, dest=%d", srcSel, destSel)
+			}
+			for _, sender := range update.AddedAllowlistedSenders {
+				if sender == (common.Address{}) {
+					return fmt.Errorf("can't allowlist 0-address sender for src=%d, dest=%d", srcSel, destSel)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func UpdateOnRampAllowListChangeset(e deployment.Environment, cfg UpdateOnRampAllowListConfig) (deployment.ChangesetOutput, error) {
+	if err := cfg.Validate(); err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	onchain, err := LoadOnchainState(e)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	var batches []timelock.BatchChainOperation
+	timelocks := make(map[uint64]common.Address)
+	proposers := make(map[uint64]*gethwrappers.ManyChainMultiSig)
+	ctx := e.GetContext()
+	for srcSel, updates := range cfg.UpdatesByChain {
+		txOps := e.Chains[srcSel].DeployerKey
+		//nolint:fatcontext // context is stored outside the loop
+		txOps.Context = ctx
```

Comment:
I believe the deployer key defines its own context, do we need to perform this assignment?